### PR TITLE
[CALCITE-4150] Fix NULL type handling in RelToSqlConverter

### PR DESCRIPTION
**Pull request by anha1**
https://issues.apache.org/jira/browse/CALCITE-4150

### CODE REVIEW

**Comment by amaliujia** [comment]
This change looks good!

@anha1 Can you also verify whether there is an equivalent test case in Sql to Rel conversion (e.g. SqlToRelConveterTest)?

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 1]
```diff
@@ -5050,6 +5050,13 @@ private void checkLiteral2(String expression, String expected) {
   }
 
   @Test void testSelectNull() {
+    String query = "SELECT NULL AS DUMMY FROM \"product\"";
+    final String expected = "SELECT CAST(NULL AS NULL) AS \"DUMMY\"\n"
+            + "FROM \"foodmart\".\"product\"";
+    sql(query).ok(expected);
```
Comment (danny0405) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:5056]: Can we promote the `CAST(NULL AS NULL)` to just `NULL` ?

Comment (anha1) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:5063]: Thanks: I missed this improvement using 1.23 as a reference. fixed.

**Comment by anha1** [comment]
> This change looks good!
> 
> @anha1 Can you also verify whether there is an equivalent test case in Sql to Rel conversion (e.g. SqlToRelConveterTest)?

Tests added into SqlToRelConverterTest.

File: core/src/main/java/org/apache/calcite/sql/type/SqlTypeUtil.java [hunk 2]
```diff
@@ -993,7 +993,7 @@ public static SqlDataTypeSpec convertTypeToSpec(RelDataType type,
     assert typeName != null;
 
     final SqlTypeNameSpec typeNameSpec;
-    if (isAtomic(type)) {
+    if (isAtomic(type) || isNull(type)) {
```
Comment (danny0405) [inline core/src/main/java/org/apache/calcite/sql/type/SqlTypeUtil.java:996]: This change seems not necessary any more ?

Comment (anha1) [reply core/src/main/java/org/apache/calcite/sql/type/SqlTypeUtil.java:996]: I would like to keep it as far as it fixes and issue in a public utility method. But thank you for spotting that SQL tests no longer hit it, so I covered this fix in SqlTypeUtilTest.

**Comment by amaliujia** [comment]
+1 LGTM 

Thanks for your contribution!

**Pull request closed without merging**